packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -261,8 +261,10 @@ library LibProposing {
             }
 
             // Check that after hooks, the Taiko Token balance of this contract
-            // have increased by at least config.livenessBond
-            if (tko.balanceOf(address(this)) < tkoBalance + config.livenessBond) {
+            // have increased by the same amount as config.livenessBond (to prevent)
+            // multiple draining payments by a malicious proposer nesting the same
+            // hook.
+            if (tko.balanceOf(address(this)) != tkoBalance + config.livenessBond) {
```

Comment:
Why we need to be strict on this?

Reply:
> Why we need to be strict on this?

First: Who likes to overpay ? :) 
Second: Currently the design now allows replaying the same hook.data multiple times, and then prover pays multiple times the liveness bond within the same proposeBlock(). This is preventing it.

Reply:
The provider will check the hook addresses and only sign the Assignment if all checks pass.